### PR TITLE
factorize tileMatrixSet for wmtsGetCap

### PR DESCRIPTION
**Pull request by rfrancel**
@loicgasser as explained in the related issue https://github.com/geoadmin/mf-chsdi3/issues/1956, I let you test:

test link: http://mf-chsdi3.dev.bgdi.ch/frr_tilematrixset/rest/services/api/1.0.0/WMTSCapabilities.xml

### CODE REVIEW

**Comment by loicgasser** [comment]
Thanks
What about:
https://github.com/geoadmin/mf-chsdi3/tree/master/chsdi/templates/wmtscapabilities

Tile Matrix set 18 to 28?

**Comment by rfrancel** [comment]
sorry @loicgasser I removed them..

**Comment by gjn** [comment]
as mentioned in the issue, I'd like to see a diff of the official getcap between this version and the one from this pr.

**Comment by rfrancel** [comment]
@gjin you can find a diff on this link (available a month):
https://www.diffchecker.com/6v5qj7tm
left is the official getCap, and right, the dev one.

**Comment by loicgasser** [comment]
We should keep the https links, here they have been changed to http. Nice tool I didn't know. ;)

**Comment by gjn** [comment]
Try to use `90.7143025612` for the dpi, should be close then, I think.

**Comment by rfrancel** [comment]
Here is a new diff with a https request and dpi 90.7143025612. It's closer.
https://www.diffchecker.com/qeothrcw

**Comment by gjn** [comment]
Better with `90.7142857739`?

**Comment by rfrancel** [comment]
yes it's way better with dpi 90.7143025612.
For example:

	<ows:Identifier>7</ows:Identifier>
current values:
<ScaleDenominator>8035734.69647</ScaleDenominator>
with dpi 90.7:
<ScaleDenominator>8034432.75</ScaleDenominator>
with dpi 90.7143025612:
<ScaleDenominator>8035699.70663</ScaleDenominator>

     <ows:Identifier>26</ows:Identifier>
current values:
<ScaleDenominator>1785.71882144</ScaleDenominator>
with dpi 90.7:
<ScaleDenominator>1785.4295</ScaleDenominator>
with dpi 90.7143025612:
<ScaleDenominator>1785.71104592</ScaleDenominator>

File: chsdi/templates/wmtscapabilities/TileMatrixSet_21781.mako [hunk 1]
```diff
@@ -1,126 +1,28 @@
...
-        <%include file="TileMatrixSet_23.mako"/>
-        <%include file="TileMatrixSet_24.mako"/>
-        <%include file="TileMatrixSet_25.mako"/>
-        <%include file="TileMatrixSet_26.mako"/>
-        <%include file="TileMatrixSet_27.mako"/>
-        <%include file="TileMatrixSet_28.mako"/>
-    </TileMatrixSet>
+<%
+  from gatilegrid import GeoadminTileGrid
+  minZoom = 0
```
Comment (loicgasser) [inline chsdi/templates/wmtscapabilities/TileMatrixSet_21781.mako]: Not sure you're using those.

**Comment by loicgasser** [comment]
@rfrancel What about

https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/templates/wmtscapabilities/wmtscapabilities.mako#L141 ?

Ideally I'd like to have at least code as possible in the Makos. (harder to debug and test)

Just pass a new object here:
https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/views/wmtscapabilities.py#L79

And please don't forget to add the related tests I mentionned here:
https://github.com/geoadmin/mf-chsdi3/issues/1223

**Comment by rfrancel** [comment]
@loicgasser I adapted the 2056 and 21781 makos so that an include line with arguments is enough to call the different getCap.

**Comment by loicgasser** [comment]
Can you provide a test link please? I think it looks good.

**Comment by gjn** [comment]
@rfrancel Status of this? And please provide link to latest diff between old and new.

And maybe you will also find the magic dpi...

**Comment by rfrancel** [comment]
I changed the dpi to dpi=90.71469755842011 as mentioned in discussion : https://github.com/geoadmin/mf-chsdi3/pull/1980
Here is the diff:
https://www.diffchecker.com/rihw8b0n
Abstract:
current getCap:
<ows:Identifier>0</ows:Identifier>
<ScaleDenominator>14285750.5715</ScaleDenominator>
<ows:Identifier>17</ows:Identifier>
<ScaleDenominator>357143.764288</ScaleDenominator>

new getCap: 
<ows:Identifier>0</ows:Identifier>
<ScaleDenominator>14285750.5715</ScaleDenominator>
<ows:Identifier>17</ows:Identifier>
<ScaleDenominator>357143.764287</ScaleDenominator>

I deployed my test branch: http://mf-chsdi3.dev.bgdi.ch/frr_tilematrixset/rest/services/api/1.0.0/WMTSCapabilities.xml

**Comment by loicgasser** [comment]
Looks like we removed a lot of lines. Good and result is the same.

**Comment by procrastinatio** [comment]
I did test for content in QGIS and Web client for EPSG 21781, 2056, 3857 and 4326. Everything looks OK.

See_ http://mf-chsdi3.dev.bgdi.ch/frr_tilematrixset/1.0.0/WMTSCapabilities.EPSG.{EPSG}.xml

with _{EPSG}_ one of the above mentionned code.

**Comment by loicgasser** [comment]
Thank you testing. @gjn this is good to merge

**Comment by gjn** [comment]
@rfrancel the dpi hunter.

I like this PR, good refactoring. Thanks alot for the patience and the testing.

**Pull request merged**